open-api/rest-catalog-open-api.yaml

```diff
@@ -3405,6 +3490,101 @@ components:
       allOf:
         - $ref: '#/components/schemas/ScanTasks'
 
+    GetEventsRequest:
+      type: object
+      properties:
+        page-token:
+          $ref: "#/components/schemas/PageToken"
+        page-size:
+          type: integer
+          format: int32
+          description: >
+            The maximum number of events to return in a single response.
+            If not provided, the server may choose a default page size.
+            Servers may return less results than requested for various reasons, such as
+            server side limits, payload size or processing time.
+        after-timestamp-ms:
+          type: integer
+          format: int64
+          description: >
+            The (server) timestamp in milliseconds to start consuming events from (inclusive).
+            If not provided, the first available timestamp is used.
+        operation-types:
+          type: array
+          items:
+            $ref: "#/components/schemas/OperationType"
+          description: >
+            Filter events by the type of operation.
+            If not provided, all types are returned.
+        catalog-objects:
+          type: array
+          discriminator:
```

Comment:
This is probably my lack of experience with Open API, but what is this doing? Is the array of a single type or multiple types?

My expectation is that `items` would be a type that has a descriminator so you could request changes for a view and a table -- for instance to see changes to a view that references a table. But with the descriminator nested in the array I'm wondering if this is a single type for all items. And if so, what does that look like in JSON given that the array itself must contain objects and can't have a `reference-type` property.

Reply:
Looks like `reference-type` is included in each type definition so I'm guessing this behaving as I would expect. It just seems like a confusing way to embed this since the array can't have a property. Refactoring this may make it more clear.

Reply:
Your understanding of the behavior is correct. I tested it with two code generators and both worked OK. Nonetheless it is a quite exotic construct. I used it to avoid introducing another type and safe some code.

As clarity is more important, I added a new type `CatalogObjectReference` now.